x-pack/plugins/apm/server/lib/services/get_service_agent.ts

```diff
@@ -69,6 +59,14 @@ export async function getServiceAgent({
           ],
         },
       },
+      aggs: {
+        runtimeName: {
+          terms: {
+            field: SERVICE_RUNTIME_NAME,
+            size: 1,
+          },
+        },
+      },
```

Comment:
Why is this agg needed? Can't you just get it from source?

Reply:
There's no guarantee that the document returned will have the `service.runtime.name` field. For instance if you run this query on dev cluster, you'll see a lot of documents without this field:

```
GET apm-7.14.0-metric-000001/_search
{
  "_source": [
    "service.runtime.name",
    "agent.name"
  ],
  "query": {
    "bool": {
      "filter": [
        {
          "term": {
            "service.name": "opbeans-ruby"
          }
        },
        {
          "range": {
            "@timestamp": {
              "lte": "now-15m"
            }
          }
        }
      ],
      "must_not": [
        {
          "exists": {
            "field": "service.runtime.name"
          }
        }
      ]
    }
  }
}
```

Reply:
I talked about that during the tech session with the team and I'm replacing it for a `should` query which will increase the score of documents that have the `service.runtime.name` field.